public/scripts/index.js

```diff
@@ -1,13 +1,72 @@
-(function (){
+function getWordFormData(form) {
+    let wordData = {}
+    //get content, reading example
+    wordData.content = document.getElementById('word-content').value
+    wordData.phoneticReading = document.getElementById('word-reading').value
+    wordData.example = document.getElementById('word-example').value
+    //get meanings
+    wordData.meanings = Array.from(document.getElementById('word-meanings').querySelectorAll('input')).map((word) => { return word.value })
+
+    //get knowledge level
+    let reducer = (accumulator, currentValue) => {
+        if (currentValue.checked) {
+            return accumulator + parseInt(currentValue.value)
+        } else { return accumulator }
```

Comment:
@evolveris has the same problem sometimes. Only check what is conditional, not everything:
`return accumulator + (currentValue.checked ? parseInt(currentValue.value) : 0)`